packages/syft/src/syft/core/tensor/smpc/share_tensor.py

```diff
@@ -329,8 +332,12 @@ def rmatmul(self, y: torch.Tensor) -> "ShareTensor":
         Returns:
             ShareTensor. Result of the operation.
         """
-        ShareTensor.sanity_checks(y)
-        return y.matmul(self)
+        if isinstance(y, ShareTensor):
+            raise ValueError("Private matmul not supported yet")
+
+        ShareTensor.sanity_check(y)
+        new_share = y.apply_function(self, "matmul")
```

Comment:
Q:Does this scenario, since y is ``torch.Tensor`` , it does not have ``apply_function`` method.?